JS-Algorithms/challenges.js

```diff
@@ -70,14 +104,22 @@ The last 3 digits for the sum of powers from 1 to 10 is "317"
 ***** */
 
 const ownPower = (number, lastDigits) => {
-  // YOUR CODE HERE...
+
+  let sum = BigInt(0); //initialize the sum on with bigint bc they need to be the same type
+  for (let i = 1; i <= number; i++) { // i choose a "for loop" bc i read about the other ways to do it, instead of "while loop" or "recursion" this seems to be the best bc is most "efficient" just compare the number that the user wrhite and compares to the i value, if its lower continues adding numbers untill is higher.
+  
+    sum  += BigInt(i) ** BigInt(i); // used += to reasignate the sum by itself and use big int bc the exponential numbers are big 
```

Comment:
A small suggestion, you can initialize the bigInt by doing:

```suggestion
  for (let i = 1n; i <= number; i++) { // i choose a "for loop" bc i read about the other ways to do it, instead of "while loop" or "recursion" this seems to be the best bc is most "efficient" just compare the number that the user wrhite and compares to the i value, if its lower continues adding numbers untill is higher.
  
    sum  += i ** i; // used += to reasignate the sum by itself and use big int bc the exponential numbers are big 
```